### PR TITLE
Add auth setup for private registry with scope

### PR DESCRIPTION
**Pull request by vitoorgomes**
Bun is awesome, but I've seen too many workarounds to make authentication for a private package under a registry-url (like `'https://npm.pkg.github.com/'`) with a scope

Heavily inspired by the `setup-node` action, I created this to support this feature

I've tested a lot on a private personal org with a private package and my repo that installs the private package on CI worked like a charm.

image

SCR-20230909-6r

Also `"@actions/cache"` import was duplicated on `setup.ts` and now I left only `import * as cache from "@actions/cache";`

### CODE REVIEW

File: src/auth.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+      if (!line.toLowerCase().startsWith(scope)) {
+        newContents += line + EOL;
+      }
+    });
+
+    newContents += EOL;
+  }
+
+  const bunRegistryString = `'${scope}' = { username = "${github.context.actor}", password = "$BUN_AUTH_TOKEN", url = "${registryUrl}" }`;
+
```
Comment (zerocowl) [inline src/auth.ts]: Considering good security practices and CI processes, I suggest changing the setting to token structure and not username/password and also adding a fallback if 'scope' has not been configured to use only ```[install]``` 

```suggestion
const bunRegistryString = `'${scope}' = { token = "$BUN_AUTH_TOKEN", url = "${registryUrl}"}`;

```

File: dist/auth.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        curContents.split(EOL).forEach((line) => {
+            // Add current contents unless they are setting the registry
+            if (!line.toLowerCase().startsWith(scope)) {
+                newContents += line + EOL;
+            }
+        });
+        newContents += EOL;
+    }
+    const bunRegistryString = `'${scope}' = { username = "${github.context.actor}", password = "$BUN_AUTH_TOKEN", url = "${registryUrl}" }`;
+    newContents += `[install.scopes]${EOL}${EOL}${bunRegistryString}${EOL}`;
```
Comment (zerocowl) [inline dist/auth.js:33]: add a fallback if the 'scope' is not configured to use only ``` [install] ```

**Review by zerocowl — COMMENTED** [review]
@vitoorgomes Nice job! I also have some suggestions

**Review by CyberFlameGO — CHANGES_REQUESTED** [review]
Might wanna remove `node_modules` - generally this shouldn't be committed

**Comment by xhyrom** [comment]
There are some conflicts, can you resolve them?

**Comment by vitoorgomes** [comment]
> Might wanna remove `node_modules` - generally this shouldn't be committed

when I forked `node_modules` were still a thing in this repo, gladly it was removed I also removed from this PR

**Comment by vitoorgomes** [comment]
> There are some conflicts, can you resolve them?

didn't had free time sooner, but it's done

**Comment by xhyrom** [comment]
can you rebuild it one more time, please? Then ping me, i'll merge.

**Comment by vitoorgomes** [comment]
> can you rebuild it one more time, please? Then ping me, i'll merge.

sure, done

**Pull request merged**